### ISSUE TITLE
Bigarray error when `ocamlc -where` is included 

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/ocaml#10833]
Hello, 

Sorry to distract with such mundane problems but I find the following behaviour, reproducible on at least 4.12 and 4.13 slightly puzzling: 

```
> echo 'let f = Bigarray.float32' > a.ml
> ocamlopt a.ml      # So far so good 
> rm *.cmi *.cmx *.o *.out                 
> ocamlopt -I $(ocamlc -where) a.ml     
File "a.ml", line 1:
Error: No implementations provided for the following modules:
         Bigarray referenced from a.cmx
```

Likely due to the fact that the "old" bigarray library still lives there, hence `bigarray.cmi` gets pickup. The following suceeds: 

```
ocamlopt -I $(ocamlc -where) bigarray.cmxa a.ml
```
But it's quite annoying. 

It would be nice if the distribution could isolate its libraries in dedicated directories.

**Comment by nojb** [comment]
Another solution is not to install the "old" bigarray library (not sure how to do this with OPAM though...)

**Comment by nojb** [comment]
You can also use `Stdlib.Bigarray` in your program.

**Comment by dbuenzli** [comment]
> You can also use `Stdlib.Bigarray` in your program.

That's not really a solution. Bigarray is supposed to be part of the stdlib nowadays. I don't want to change my program if I suddenly add say `unix` as a dependency. The problem is that for any library I can do 
```
> ocamlfind query -i-format $LIB
```
to get my `-I` flags but now that doesn't work if $LIB is distributed by OCaml since in particular we have: 
```
> ocamlfind query -i-format stdlib 
-I /Users/dbuenzli/.opam/4.13.0/lib/ocaml
> ocamlfind query -i-format unix
-I /Users/dbuenzli/.opam/4.13.0/lib/ocaml
```
which leads to the above problem. 

I guess the solution is rather to detect `ocamlc -where` and not include it – which of course is brittle (and annoying).

**Comment by dbuenzli** [comment]
Fixed by #10896
